### ISSUE TITLE
Add Tap Handlers to `TableRow`

### ISSUE THREAD

**Issue by escamoteur** [flutter/flutter#42609]
## Use case

While laying out a drawer menu I decided to give Table a try because it seems a good fit if you want to show icons before the entries, sometimes another icon behind in a regular grid.

When the user wants to select a menu entry it should be possible that he taps anywhere on the Row.
Adding GestureDetectors for every cell seems not a good solution

IMHO the use case to make w whole row selectable is quite common for instance highlightning a row on tap.

## Proposal

Add an OnTap handler to TableRows that adds a GestureDetector to a row is a Handler is set.

**Comment by Zazo032** [comment]
Isn't this currently achievable using `ListTile` widget? Using `leading` and `trailing` parameters to show icons. They also have gesture recognizers built in.

**Comment by escamoteur** [comment]
If I only have three columns that would be an option. Also List Tile doesn't allow to configure colum widths as flexible

**Comment by Zazo032** [comment]
That's right, `ListTile` has Material Design specifications, this proposal would be really nice for more flexible layouts by providing a kind of more generic/flexible `ListTile`.

**Comment by JayPerfetto** [comment]
Glad to see this is getting the attention it deserves from the Flutter team lol

**Comment by alexmercerind** [comment]
> @escamoteur Do you propose to have ink well affect as well when a row is tapped?

Yes.

**Comment by escamoteur** [comment]
Would probably be nice to have this optional

**Comment by AbhishekDoshi26** [comment]
Any update on this? Or atleast a workaround till the time this gets done?

**Comment by adonisRodxander** [comment]
Any update? Or some ideas of how to do this?

**Comment by adonisRodxander** [comment]
This is what I did:

```dart
class _TableRow2 implements TableRow {
  final List<Widget> tableCells;
  final Decoration? customDecoration;
  final VoidCallback? callback;

  _TableRow2({
    required this.tableCells,
    this.callback,
    this.customDecoration,
  });

  @override
  List<Widget> get children => tableCells
      .map(
        (cell) => TableRowInkWell(
          onTap: callback,
          overlayColor: WidgetStatePropertyAll(Colors.transparent),
          child: cell,
        ),
      )
      .toList();

  @override
  Decoration? get decoration => customDecoration;

  @override
  LocalKey? get key => null;
}
```

With this solution, I avoid sending the callback to each cell individually, and the `onTap` is executed for the entire row (whereas with `GestureDetector`, only the cell content responds to `onTap`).
